core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -424,6 +424,11 @@ object SparkSubmit {
       args.files = mergeFileLists(args.files, args.primaryResource)
     }
 
+    // Ignore insecure TLS certificates?
+    if (args.insecure_skip_tls_verify) {
+      System.setProperty("spark.ssl.insecureSkipTLSVerify", "true")
```

Comment:
It's only client-side. E.g. if you have a setup of DCOS with snake-oil certificates in the admin router (which is still the default setup).